examples/research_projects/xls_r/README.md

```diff
@@ -393,3 +470,99 @@ error rate of 27% as can be seen on the automatically generated [model card](htt
 The above-chosen hyperparameters probably work quite well on a range of different 
 datasets and languages, but are by no means optimal. It is up to you to find a good set of 
 hyperparameters.
+
+
+## How to finetune with OVH cloud
+
+TODO(Anton)
+
+## How to combine n gram with acoustic model
+
+Having trained a speech recognition model with CTC as shown in the section above, 
+one can further improve the model's performance by adding an **n-gram language model**
+to the decoding process of the model. By doing so, we are replacing the naive greedy decoding 
+by **n-gram-boosted** beam search decoding.
+
+N-gram language models can be built on CPU in just a few minutes. *N-gram-boosted* beam search decoding noticebly slows down the 
+inference time, but also yields significant word error rates improvements - usually between 10-40 %.
+
+You can find an in-detail blog post on how to build an *n-gram* [here](https://huggingface.co/blog/wav2vec2-with-ngram).
+The blog post can be opened in a google colab and by adapting three lines of the example for your use case, one can directly
+create an *n-gram* in the google colab.
+The blog post gives in-detail instructions on: how to built an n-gram and how to add it to your trained speech recognition model.
+
+- why one should add an *n-gram* to her/his speech recognition system,
+- how to build an *n-gram*, and,
+- how to add the built *n-gram* the speech recognition system for seamless decoding
+
+Our previously trained model - [xls-r-300m-sv](https://huggingface.co/hf-test/xls-r-300m-sv) - enjoys a 30% word error rate reduction after 
+having added an n-gram. As shown in the example of the blog post, we strongly advise participants to upload all files required for combining 
+the *n-gram* with a trained speech recognition model directly into the same model repository.
+
+## Evaluation
+
+```bash
+./eval.py --model_id hf-test/xls-r-300m-sv --dataset mozilla-foundation/common_voice_7_0 --config sv-SE --split test --log_outputs
+```
+
+works, but
+
+```bash 
+./eval.py --model_id hf-test/xls-r-300m-sv --dataset speech-recognition-community-internal/tedx_manual_dev_test --config sv --split validation --chunk_length_s 5.0 --stride_length_s 1.0 --log_outputs
```

Comment:
This is not supported yet. It's doing pipeline with chunking + language model integration. Would be great if we could make it possible to do this for the sprint. It should not be too difficult actually. We can just do the same trick with the output logits vectors instead of the argmax output ids (as done for no LM) and then decode one huge logits matrix